# 20200813 Nanopore HTT Analysis

Index Fast5
```
condor_submit -name ettore ~/strique/jobs/20200813/condor-striqueIndex_touch.job
```
BAM2SAM
```
for n in 0 1 ; do condor_submit -name ettore -a "n=$n" ~/strique/jobs/20200813/condor-bam2sam.job ; done
```

Count CAG
```
for n in 0 1 ; do condor_submit -name ettore -a "n=$n" ~/strique/jobs/20200813/condor-striqueCount.job ; done
```

## Basecalling

Guppy_basecaller for a single Qscore
```
~/src/ont-guppy-cpu/bin/./guppy_basecaller -i /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/all_fast5/ -s /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_13/ -c /lustrehome/gianluca/src/ont-guppy-cpu/data/dna_r9.4.1_450bps_hac.cfg --qscore_filtering 13
```
or Loop for each Qscore
```
for n in 7 7.5 8 8.5 9 9.5 10 13 ; do echo condor_submit -name ettore -a "qscore=$n" /lustrehome/gianluca/strique/jobs/20200813/condor-guppy_basecaller_qscore_exp.job ; done
```
## Alignment

Guppy_aligner
```
for c in guppy_qscore_7 guppy_qscore_7.5 guppy_qscore_8 guppy_qscore_8.5 guppy_qscore_9 guppy_qscore_9.5 guppy_qscore_10 guppy_qscore_13 ; do ~/src/ont-guppy-cpu/bin/./guppy_aligner -i /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/$c/pass/ -s /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/$c/alignment/ --align_ref /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/reference/pUC57_Cre_Lox_50Q.fasta ; done
```

Minimap2 aligner
```
minimap2 -y -a -x map-ont -t 2 --MD /reference_folder/exon1-amp-left.fasta.minimap2.idx /tmp/analysis/4/21_109_1_1_20190201_0947_MN29119_FAK49673_a9e8b3a2_fastq_fastq_runid_361d77ef4849a8b084ddc941d7c81aec28e5c109_10.fastq
```

## STRique analysis

STRique count all different qscore
```
for n in 7 7.5 8 8.5 9 9.5 10 13 ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_$n/alignment/all_fastq.sam" -a "fofn=/lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/all_fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/20200813/HTT_config_20200813_pUC57_Cre_150_CAG.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/pUC57-Cre/20200813_allSAM_qscore_$n.tsv" -a "n=qscore_$n" condor-striqueCount_exp.job ; done
```
STRique count with different config file for different qscore
```
for n in 7 10 13; do for c in 150_600bp_CAG 150_1000bp_CAG 200_600bp_CAG 200_1000bp_CAG 200_CAG 150_CAG ; do echo condor_submit -name ettore -a "sam=/lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_$n/alignment/all_fastq.sam" -a "fofn=/lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/all_fast5/reads.fofn" -a "config=/lustre/home/enza/cattaneo/config_file/20200813/HTT_config_20200813_pUC57_Cre_$c.tsv" -a "output=/lustrehome/gianluca/strique/striqueOutput/pUC57-Cre/20200813_allSAM_qscore$n.$c.tsv" -a "n=$c.qscore$n" ~/strique/jobs/20200813/condor-striqueCount_exp.job ; done ; done
```
### [R analysis](count_CAG_allQscore.R) of STRique OUTPUT

```
Rscript ~/projects/cattaneo/debug/count_CAG_allQscore.R
```

## DEBUG

### [R analysis](plot_summaryAlignment.R) for Debug Alignment

```
Rscript ~/projects/cattaneo/debug/plot_summaryAlignment.R
```

info from Fast5
```
# number of fast5 reads

cat /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/all_fast5/reads.fofn | wc -l
```
info from fastq reads with qscore > THRESHOLD (PASS)
```
# number of pass reads

cat /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/pass/fastq_runid_ab4c51e370a39025bfacdeabffb0e2d16507f36b_* | grep sampleid | wc -l
```
info from fastq reads with qscore < THRESHOLD (FAIL)

```
# number of fail reads

cat /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/fail/fastq_runid_ab4c51e370a39025bfacdeabffb0e2d16507f36b_* | grep sampleid | wc -l
```
info from SAM
```
# number of sam reads

cat /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/alignment/fastq_runid_ab4c51e370a39025bfacdeabffb0e2d16507f36b_* | grep -v @ | cut -f 1 | sort | uniq | wc -l
```
info from different STRique's debug_logs 
```
# Loop for all info

for t in 150_CAG 150_600bp_CAG 150_1000bp_CAG 200_CAG 200_600bp_CAG 200_1000bp_CAG ; do for c in No Test Err ; do cat $t.qscore7.count.err  | cut -f 4 -d ":" | grep $c | tr " " "\t" | cut -f 5 | wc -l ; done ; done
```
### [R analysis](plot-debug.R) for debugging using all different combination of prefix/suffix length, Ref-Interval and info about basecalling and debug_log from STRique.

```
Rscript /Users/gianlucadamaggio/projects/cattaneo/debug/plot-debug.R
```

### Merge summary file from basecalling with debug_log from STRique

#### Info from STRique's debug_log  

reads with no target for config file
```
cat qscore_7.count.err | cut -f 4 -d ":" | grep "No" | tr " " "\t" |cut -f 5 | wc -l
```
Error parsing alignment
```
cat qscore_7.count.err | cut -f 4 -d ":" | grep "Err" | tr " " "\t" |cut -f 5 | wc -l
```
Tested for targets
```
cat qscore_7.count.err | cut -f 4 -d ":" | grep "Test" | tr " " "\t" |cut -f 5 | wc -l
```
list of unique IDs
```
cat qscore_7.count.err | cut -f 4 -d ":" | grep "No" | tr " " "\t" |cut -f 5 | sort | uniq  > uniq_id.txt
```
list of reads used by strique
```
cat qscore_7.count.err | cut -f 4 -d ":" | grep "Test" | tr " " "\t"|cut -f 3 > good_reads.txt
```
#### [R analysis](tested4targetsXsummaryAlign.R) after obtain reads used by STRique (good_reads)
```
Rscript ~/projects/cattaneo/debug/tested4targetsXsummaryAlign.R
```



## Create a pangenome from fast5 ( IN DEVELOPMENT )

### minimap2
```
for n in 7 10 ; do condor_submit -name ettore -a "qscore = $n" /lustrehome/gianluca/jobs/cattaneo/pangenome/condor-minimap2_matrixAllReads.job ; done
```
```
/lustrehome/gianluca/src/anaconda3/bin/minimap2 -t 16 -c -X /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_$1/pass/all_reads.fastq /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_$1/pass/all_reads.fastq | /bin/gzip > /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_$1/pass/qscore$1.paf.gz
```

### seqwish
```
for n in 7 10 ; do condor_submit -name ettore -a "qscore = $n" /lustrehome/gianluca/jobs/cattaneo/pangenome/condor-seqwish.job ; done
```
```
/lustrehome/gianluca/src/anaconda3/bin/seqwish -s /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/pass/all_reads.fastq -p <(zcat /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/pass/qscore7.paf.gz /lustrehome/gianluca/src/anaconda3/bin/fpa drop -l 2000) -g /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.gfa -P
```
### odgi
```
odgi build -g /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.gfa -o - | odgi sort -i - -o - -p Ygs -t 16 -P > /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.sort.odgi
odgi viz -i /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.sort.odgi -o /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.sort.odgi.png -x 4000 -y 500 -R
odgi view -i /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.sort.odgi -g > /lustre/home/enza/cattaneo/data/20200813_1444_MN29119_AEG653_694c71a4/new_basecalling/guppy_qscore_7/gfa/qscore7.sort.gfa
```


minimap2 -t 16 -c -X $f $f |gzip > HLA/$b.paf.gz
seqwish -s HLA/$b.fa.gz -p <(zcat HLA/$b.paf.gz fpa drop -l 2000) -g HLA-gfas/$b.gfa -P
odgi build -g HLA-gfas/$b.gfa -o - | odgi sort -i - -o - -p Ygs -t 16 -P > HLA-gfas/$b.sort.odgi
odgi viz -i HLA-gfas/$b.sort.odgi -o HLA-gfas/$b.sort.odgi.png -x 4000 -y 500 -R
odgi view -i HLA-gfas/$b.sort.odgi -g > HLA-gfas/$b.sort.gfa
